proxy/start_container_interceptor.go

```diff
@@ -0,0 +1,82 @@
+package proxy
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/fsouza/go-dockerclient"
+	. "github.com/weaveworks/weave/common"
+)
+
+type startContainerInterceptor struct {
+	client  *docker.Client
+	withDNS bool
+}
+
+func StartContainerInterceptor(client *docker.Client, withDNS bool) Interceptor {
+	return &startContainerInterceptor{
+		client:  client,
+		withDNS: withDNS,
+	}
+}
+
+func (i *startContainerInterceptor) InterceptRequest(r *http.Request) (*http.Request, error) {
+	return r, nil
+}
+
+func (i *startContainerInterceptor) InterceptResponse(res *http.Response) (*http.Response, error) {
+	containerId := containerFromPath(res.Request.URL.Path)
+
+	container, err := i.client.InspectContainer(containerId)
+	if err != nil {
+		Warning.Print("Error Inspecting Container: ", err)
+		return res, nil
+	}
+
+	if cidrs, ok := weaveCidrsFromConfig(container.Config); ok {
+		Info.Printf("Container %s was started with CIDR \"%s\"", containerId, strings.Join(cidrs, " "))
+		if err := i.attachContainerToWeave(containerId, cidrs); err != nil {
+			Warning.Print("Attaching container to weave failed: ", err)
+			return res, nil
+		}
+	} else {
+		Debug.Print("No Weave CIDR, ignoring")
+	}
+
+	return res, nil
+}
+
+func containerInfo(transport http.RoundTripper, containerId string) (body map[string]interface{}, err error) {
+	body = nil
+	client := &http.Client{
+		Transport: transport,
+	}
+	if res, err := client.Get("http://localhost/v1.16/containers/" + containerId + "/json"); err == nil {
```

Comment:
that version number should not be hard-coded here. should be a constant at least, but ideally it would be set at compile time.